src/hooks/use-slides.tsx

```diff
@@ -20,23 +22,40 @@ export function useCollectSlides() {
     ) as unknown as Iterable<HTMLElement>;
 
     const nextSlideIds: SlideId[] = [];
+    const nextSlideIdsOfSlidesWithTemplates: Set<SlideId> = new Set();
     for (const placeholderNode of slides) {
-      const { slideId } = placeholderNode.dataset;
+      const { slideId, slideHasTemplate } = placeholderNode.dataset;
       nextSlideIds.push(slideId!);
+      if (slideHasTemplate === 'true') {
+        nextSlideIdsOfSlidesWithTemplates.add(slideId!);
```

Comment:
I know there's an existing `slideId!` (with the postfix `!` annotation), so it makes sense to just do the same here. In general, I feel pretty strongly against the `!` annotation though, because it limits the utility of TS (we're saying we _know_ the value is defined, even if we haven't actually checked that it is). 

Perhaps in a future PR, we can address that issue and clean up both of these `slideId!` instances, just by doing a check that `slideId !== undefined`.